stl/inc/xstring

```diff
@@ -2943,9 +2942,9 @@ private:
         // NOTE: even if _Right is in large mode, we only go into large mode ourselves if the actual size of _Right
         // requires it
         if (_Stay_small) { // stay small, don't allocate
-            _Traits::copy(_My_data._Bx._Buf, _Right_ptr, _BUF_SIZE);
-            _My_data._Mysize = _Right_size;
-            _My_data._Myres  = _BUF_SIZE - 1;
+            _Traits::copy(_My_data._Bx._Buf, _Right_data._Bx._Buf, _BUF_SIZE);

```

Comment:
This is where we save extra branch, instead of `_Right_data._Myptr()` we use SSO bufer.
But if source for runtime string could be constexpr string, then this won't work

Reply:
Irrelevant to constexpr string, runtime string could also be small but allocated, so should use `_Myptr()` here